### PR TITLE
Hotfix/aftereach sa cleanup

### PR DESCRIPTION
**Pull request by asanzgom**

### CODE REVIEW

File: tests/rhosak/kas_messaging.spec.ts [hunk 1]
```diff
@@ -96,19 +96,24 @@ test.beforeEach(async ({ page }) => {
   expect(producerResponse === true).toBeTruthy();
 });
 
-test.afterAll(async ({ page }) => {
-  await navigateToKafkaList(page);
-
+test.afterEach(async ({ page }) => {
+  console.log('ENTRANDO EN EL AFTEREACH');
```
Comment (kornys) [inline tests/rhosak/kas_messaging.spec.ts:100]: remove all your logging info for your debugging

**Review by kornys — DISMISSED** [review]
see comments

File: tests/rhosak/kas_messaging.spec.ts (hunk 1 above)
Comment (asanzgom) [reply tests/rhosak/kas_messaging.spec.ts]: removed. Can I resolve this conversation @kornys ?

File: lib/sa.ts [hunk 2]
```diff
@@ -49,7 +49,7 @@ export const createServiceAccount = async function (page: Page, name: string) {
 };
 
 export const deleteServiceAccount = async function (page: Page, name: string) {
-  const saLinkSelector = page.getByText(name, { exact: true });
+  const saLinkSelector = page.getByText(name);
```
Comment (mmusil) [inline lib/sa.ts:52]: What was wrong with `exact`?

Comment (asanzgom) [reply lib/sa.ts:52]: image

For some reason, the exact is rigid that on that method that `saLinkSelector.count()` was evaluating to "0" avoiding to delete Service Account. I was able to see that on debugging

Comment (Frawless) [reply lib/sa.ts:52]: maybe use something better than `getByText` would make it more robust as well

File: tests/rhosak/kas_messaging.spec.ts [hunk 3]
```diff
@@ -96,19 +96,20 @@ test.beforeEach(async ({ page }) => {
...
 });
 
-test.afterAll(async ({ page }) => {
-  await navigateToKafkaList(page);
-
+test.afterEach(async ({ page }) => {
+  await navigateToSAList(page);
   try {
-    await deleteKafkaInstance(page, testInstanceName);
+    await deleteServiceAccount(page, testServiceAccountName);
```
Comment (Frawless) [inline tests/rhosak/kas_messaging.spec.ts:102]: this is already in beforeEach, what is the difference as you keep both logic?

Comment (asanzgom) [reply tests/rhosak/kas_messaging.spec.ts:102]: we have to clean up after, the one that is on beforeEach it is supposedly cleaning the one using the same name, which does not make sense anymore as we are using session id and the name will differ
```
if ((await page.locator('tr', { hasText: testServiceAccountName }).count()) !== 0) {
    await deleteServiceAccount(page, testServiceAccountName);
  }
```
I think I should delete those lines

File: lib/sa.ts (hunk 2 above)
Comment (asanzgom) [reply lib/sa.ts:52]: replaced for `const saLinkSelector = page.locator('td', { hasText: `${name}` });`

File: tests/rhosak/kas_messaging.spec.ts (hunk 3 above)
Comment (asanzgom) [reply tests/rhosak/kas_messaging.spec.ts:99]: Can I resolve this conversation @Frawless ?

File: lib/sa.ts (hunk 2 above)
Comment (asanzgom) [reply lib/sa.ts]: Can I resolve this conversation @Frawless @mmusil ?

**Review by kornys — APPROVED** [review]

**Review by Frawless — APPROVED** [review]

**Pull request merged**